src/protocols/secure_channel/CASESession.cpp

```diff
@@ -2140,10 +2256,16 @@ CHIP_ERROR CASESession::OnMessageReceived(ExchangeContext * ec, const PayloadHea
 #endif // CONFIG_BUILD_FOR_HOST_UNIT_TEST
 
 #if CHIP_CONFIG_SLOW_CRYPTO
-    if (msgType == Protocols::SecureChannel::MsgType::CASE_Sigma1 || msgType == Protocols::SecureChannel::MsgType::CASE_Sigma2 ||
-        msgType == Protocols::SecureChannel::MsgType::CASE_Sigma2Resume ||
-        msgType == Protocols::SecureChannel::MsgType::CASE_Sigma3)
-    {
+    if ((msgType == Protocols::SecureChannel::MsgType::CASE_Sigma1 || msgType == Protocols::SecureChannel::MsgType::CASE_Sigma2 ||
+         msgType == Protocols::SecureChannel::MsgType::CASE_Sigma2Resume ||
+         msgType == Protocols::SecureChannel::MsgType::CASE_Sigma3) &&
+        mExchangeCtxt.Value()->GetSessionHandle()->AsUnauthenticatedSession()->GetPeerAddress().GetTransportType() !=
+            Transport::Type::kTcp)
+    {
+        // TODO: Rename FlushAcks() to something more semantically correct and
```

Comment:
Please make sure a followup issue is filed to track fixing up this stuf and have this code comment point to that issue.

Reply:
Filed issue #33183